rs/sns/governance/canister/governance.did

```diff
@@ -21,6 +21,134 @@ type Action = variant {
   ExecuteGenericNervousSystemFunction : ExecuteGenericNervousSystemFunction;
   ManageLedgerParameters : ManageLedgerParameters;
   Motion : Motion;
+  RegisterExtension : RegisterExtension;
+  DeregisterExtension : DeregisterExtension;
+  ExecuteExtensionFunction : ExecuteExtensionFunction;
```

Comment:
We cannot reuse `ExecuteGenericNervousSystemFunction`, as that is a low-level API. It assumes there is exactly one function being called with the provided literal argument (encoded as a blob). SNS extensions should support more complex custom functions, which can involve, e.g., setting up an allowance (by the SNS Gov) and then calling some number of functions on the extension canister (e.g., notify_deposit).